Automattic/pocket-casts-android
# [Paywall redesign] Add the new paywall

## Description
- This adds the first changes for the paywall redesign. I will address the feature cards in a later PR and also the text below of the button
- See: SEQe6JBDCeegcpa9Bfg5Ww-fi-3_756
- The background will be the same for all themes: p1723556268631729-slack-C07G1MH890E 
- This redesign is currently behind a temporary local feature flag. We are going to change in a later PR for the testing experiment

Fixes #2630

## Testing Instructions
1. Apply this [paywall.patch](https://github.com/user-attachments/files/16612993/paywall.patch)
2. Run in `debug`
3. Make sure PAYWALL_EXPERIMENT feature flag is enabled
4. Make sure you have the same account logged in Play Store
5. Go to Profile -> Settings -> Pocket Casts Plus
6. Verify the Paywall redesign
7. ✅ Make sure the background is black
8. ✅ Make sure you `Not now` button instead of arrow
9. ✅ Make sure you don't see the Monthly and Yearly pills

## Screenshots or Screencast 

<img src="https://github.com/user-attachments/assets/c80f6df7-55b6-4e46-a76a-08b5f6390cf6" width="300">



## Checklist
 
#### I have tested any UI changes...
